service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -170,4 +195,162 @@ void updatePipelineWorkspaceId() {
     // assert the workspace id has been updated
     assertEquals(savedWorkspaceId, p.getWorkspaceId());
   }
+
+  // input validation tests
+  private static Stream<Arguments> inputValidations() {
+    return Stream.of(
+        // arguments: isRequired, inputs, shouldPassValidation, expectedErrorMessage
+        arguments(
+            new LinkedHashMap<String, Object>(
+                Map.of("multi_sample_vcf", "this/is/a/vcf/path.vcf.gz", "new_integer_input", 123)),
```

Comment:
it would probably be a little clearer from a mental gymnastics standpoint and be easier to update if the keys used here were pulled out into variables and used to format the error messages below

Reply:
let me know if this update isn't what you had in mind